File: crates/nil-payload/src/military.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use nil_core::military::maneuver::ManeuverRequest;
use nil_core::world::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestManeuverRequest {
  pub world: WorldId,
  pub request: ManeuverRequest,
}

File: app/src/types/request/military.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { WorldId } from '@/types/core/world';
import type { ManeuverRequest } from '@/types/core/military/maneuver';

export interface RequestManeuverRequest {
  readonly world: WorldId;
  readonly request: ManeuverRequest;
}
